File: cpp/2401.longest-nice-subarray.cpp
#include <vector>
using std::vector;

class Solution {
public:
  int longestNiceSubarray(vector<int> &nums) {
    int mask = 0xffffffff;
    int maxLen = 0;

    int left = 0;

    for (int right = 0; right < nums.size(); right++) {
      mask ^= nums[right];

      while (mask & nums[right]) {
        mask ^= nums[left++];
      }

      maxLen = std::max(maxLen, right - left + 1);
    }

    return maxLen;
  }
};

File: golang/2401.longest-nice-subarray.go
func longestNiceSubarray(nums []int) int {
	mask := 0xffffffff
	maxLen := 0
	left := 0

	for right := 0; right < len(nums); right++ {
		mask ^= nums[right]

		for mask&nums[right] > 0 {
			mask ^= nums[left]
			left++
		}

		maxLen = max(maxLen, right-left+1)
	}

	return maxLen
}
